src/test/java/seedu/address/logic/commands/EditIngredientCommandTest.java

```diff
@@ -0,0 +1,188 @@
+
+    @Test
+    public void execute_allFieldsSpecifiedUnfilteredList_success() {
+        Ingredient editedIngredient = new IngredientBuilder().build();
+        EditIngredientCommand.EditIngredientDescriptor descriptor =
+                new EditIngredientDescriptorBuilder(editedIngredient).build();
+        EditIngredientCommand editIngredientCommand = new EditIngredientCommand(INDEX_FIRST_INGREDIENT,
+                descriptor);
+
+        String expectedMessage = String.format(EditIngredientCommand.MESSAGE_EDIT_INGREDIENT_SUCCESS,
+                editedIngredient);
+
+        Model expectedModel = new ModelManager(new WishfulShrinking(model.getWishfulShrinking()), new UserPrefs());
+        expectedModel.setIngredient(model.getFilteredIngredientList().get(0), editedIngredient);
+
+        assertCommandSuccess(editIngredientCommand, model, expectedMessage, expectedModel);
+    }
+
+    @Test
+    public void execute_someFieldsSpecifiedUnfilteredList_success() {
+        Index indexLastIngredient = Index.fromOneBased(model.getFilteredIngredientList().size());
+        Ingredient lastIngredient = model.getFilteredIngredientList().get(indexLastIngredient.getZeroBased());
+
+        IngredientBuilder ingredientInList = new IngredientBuilder(lastIngredient);
+        Ingredient editedIngredient = ingredientInList.withValue(VALID_INGREDIENT_MARGARITAS)
+                .build();
+
+        EditIngredientCommand.EditIngredientDescriptor descriptor =
+                new EditIngredientDescriptorBuilder().withIngredient(
+                        new Ingredient(VALID_INGREDIENT_MARGARITAS)).build();
+        EditIngredientCommand editIngredientCommand = new EditIngredientCommand(indexLastIngredient, descriptor);
+
+        String expectedMessage = String.format(EditIngredientCommand.MESSAGE_EDIT_INGREDIENT_SUCCESS,
+                editedIngredient);
+
+        Model expectedModel = new ModelManager(new WishfulShrinking(model.getWishfulShrinking()), new UserPrefs());
+        expectedModel.setIngredient(lastIngredient, editedIngredient);
+
+        assertCommandSuccess(editIngredientCommand, model, expectedMessage, expectedModel);
+    }
+
+    @Test
+    public void execute_noFieldSpecifiedUnfilteredList_success() {
+        EditIngredientCommand editIngredientCommand = new EditIngredientCommand(INDEX_FIRST_INGREDIENT,
+                new EditIngredientCommand.EditIngredientDescriptor());
+        Ingredient editedIngredient = model.getFilteredIngredientList().get(INDEX_FIRST_INGREDIENT.getZeroBased());
+
+        String expectedMessage = String.format(EditIngredientCommand.MESSAGE_NOT_EDITED,
+                editedIngredient);
+
+        Model expectedModel = new ModelManager(new WishfulShrinking(model.getWishfulShrinking()), new UserPrefs());
+
+        assertCommandSuccess(editIngredientCommand, model, expectedMessage, expectedModel);
+    }
+
+    //@Test
```

Comment:
This test can't work?

Reply:
yeah not very sure why. I'll fix the relevant commented out tests in the next iteration.